contracts/DomainRegistryImplementationV2.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./DomainUtils.sol";
+
+contract DomainRegistryImplementationV2 is Initializable, OwnableUpgradeable {
+
+    using Address for address payable;
+    using DomainUtils for string;
+
+    uint256 public domainPrice; 
+
+    struct Domain {
+        bool isRegistered;
+        address owner;
+    }
+
+    mapping(string => Domain) public domains;
+    mapping(string => uint256) public rewards;
```

Comment:
`rewards` and `domainRewards` naming is confusing it says nothing to whoever reads it. Try to figure out better naming.   -1 point